### PR TITLE
Replace Watchdog to fix celery error in dev

### PR DESCRIPTION
**Pull request by kilbergr**
[This issue](https://github.com/harvard-lil/capstone/issues/2094) brought to light an issue with using Watchmedo (Watchdog) to auto reload the celery worker for elastic search indexing. Unfortunately, this is a [known issue](https://github.com/gorakhargosh/watchdog/issues/838) with the current solution of switching to `arm64-capable` (we're running `linux/amd64`). Changing that would mean changing our regex library, which is a lot of work! Instead, I found you could use the Django's `autoreload` built in utility. 

I used this as a reference to create the command: 
https://github.com/fceruti/django-starter-project/blob/fa22e6b2496aa9943d50d9690143a2b4c9bac3be/apps/misc/management/commands/celery_autoreload.py

This PR removes the watchdog and argh requirements, since they're no longer used.

### CODE REVIEW

**Comment by matteocargnelutti** [comment]
@kilbergr Nice 😎 ! This looks good to me, but I am also not very familiar with this stack. I would suggest asking @bensteinberg for a second opinion before merging.

**Review by matteocargnelutti — APPROVED** [review]

**Review by bensteinberg — COMMENTED** [review]
I think our idiom for this kind of command would be to make it a `fab` task; see [fabfile.py](https://github.com/harvard-lil/capstone/blob/develop/capstone/fabfile.py) -- though the idea of switching to Invoke has come up in H2O: https://github.com/harvard-lil/h2o/issues/1810

**Review by rebeccacremona — CHANGES_REQUESTED** [review]
Good idea!

This is the first time I've seen custom Django management commands, so I pulled this down to play with it. 

It looks to me like it isn't quite working as intended yet: here's the output I get, subsequent to calling `fab run`:

```
 % docker compose exec web fab run                             
Starting: python manage.py run_celery_worker
[localhost] local: python manage.py runserver 0.0.0.0:8000 --nostatic
Starting celery worker with autoreload...
Performing system checks...

System check identified no issues (0 silenced).
Starting celery worker with autoreload...
Restarting celery...
kill: (165): No such process
February 13, 2023 - 15:49:23
Django version 3.2.17, using settings 'config.settings'
Starting development server at http://0.0.0.0:8000/
Quit the server with CONTROL-C.
...
[2023-02-13 15:49:35,504: INFO/MainProcess] celery@c038fde60f39 ready.
[2023-02-13 15:49:45,666: INFO/Beat] beat: Starting...
```

I see the log line from `call_subprocess` (`Starting: python manage.py run_celery_worker`), but I also see -two- commands about starting celery, plus an unwanted restart and an unsuccessful `kill`.

Then, I saved a comment into capweb/forms.py. After a second or two:
```
Performing system checks...

Starting celery worker with autoreload...
Restarting celery...
kill: (228): No such process
System check identified no issues (0 silenced).
February 13, 2023 - 15:52:51
Django version 3.2.17, using settings 'config.settings'
Starting development server at http://0.0.0.0:8000/
Quit the server with CONTROL-C.
...
[2023-02-13 15:53:02,032: INFO/MainProcess] celery@c038fde60f39 ready.
```

I again see an unsuccessful `kill`. 

When I control+c to quit, I see a shutdown from `open_subprocess` as expected.
```
^CKilling: python manage.py run_celery_worker

Stopped.

worker: Hitting Ctrl+C again will terminate all running tasks!

worker: Warm shutdown (MainProcess)
[2023-02-13 16:11:26,509: INFO/MainProcess] beat: Shutting down...
```

To make this simpler to look at, I temporarily changed the fab task to:
```
    with open_subprocess("python manage.py run_celery_worker"):
        while True:
            pass
``` 
so that the dev server doesn't interfere: it works the same way by itself:

```
% docker compose exec web fab run
Starting: python manage.py run_celery_worker
Starting celery worker with autoreload...
Starting celery worker with autoreload...
Restarting celery...
kill: (666): No such process
...
[2023-02-13 16:14:02,417: INFO/MainProcess] celery@c038fde60f39 ready.
[2023-02-13 16:14:13,207: INFO/Beat] beat: Starting...
```

I think it's probably worth spending some more time teasing out what's going on here, even if the end result seems to be that celery is restarting on code changes. 

I think probably the nested subproccess situation is a getting a little tangled.

Also, I checked to see, and `pkill` is available: that might be a little [more elegant than the grep move](https://testdriven.io/courses/django-celery/auto-reload/)?

**Comment by kilbergr** [comment]
OK what I've discovered:
 When you first run the `run_celery_worker` command, it says "I'm starting with reloader!" but actually it's not. Instead, we're getting [this](https://github.com/django/django/blob/6bdc3c58b65eb32fd63cd41849f00a17a36b4473/django/utils/autoreload.py#L663)--a check for the `DJANGO_AUTORELOAD_ENV` var (which is actually a check for [the `RUN_MAIN` var](https://github.com/django/django/blob/6bdc3c58b65eb32fd63cd41849f00a17a36b4473/django/utils/autoreload.py#L27)). When that fails, the next step is to run [restart_with_reloader](https://github.com/django/django/blob/6bdc3c58b65eb32fd63cd41849f00a17a36b4473/django/utils/autoreload.py#L270), which sets the AUTORELOAD env var to true and reruns the original command, which repeats "I'm starting with reloader!" I tried to get around it by changing the os env vars, but it wasn't as simple as adding that to our settings file and I don't want to mess with `RUN_MAIN`, which has other purposes. Instead, I took that announcement out but added a comment in the code. You still get the killing and starting announcements.

Now this is how things look when you run `fab run`:

```
root@0d3e693c3dc8:/app# fab run
Starting: python manage.py run_celery_worker
[localhost] local: python manage.py runserver 0.0.0.0:8000 --nostatic
Performing system checks...

System check identified no issues (0 silenced).
Kill lingering celery worker...
Start celery worker...
February 15, 2023 - 19:12:27
Django version 3.2.17, using settings 'config.settings'
Starting development server at http://0.0.0.0:8000/
Quit the server with CONTROL-C.

 -------------- celery@0d3e693c3dc8 v4.3.0 (rhubarb)
---- **** -----
--- * ***  * -- Linux-5.15.49-linuxkit-x86_64-with-debian-10.13 2023-02-15 19:12:36
-- * - **** ---
- ** ---------- [config]
- ** ---------- .> app:         config:0x4007e3b550
- ** ---------- .> transport:   redis://redis:6379//
- ** ---------- .> results:     redis://redis/
- *** --- * --- .> concurrency: 1 (prefork)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** -----
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery


[2023-02-15 19:12:37,230: INFO/MainProcess] Connected to redis://redis:6379//
[2023-02-15 19:12:37,289: INFO/MainProcess] mingle: searching for neighbors
[2023-02-15 19:12:38,416: INFO/MainProcess] mingle: all alone
[2023-02-15 19:12:38,469: WARNING/MainProcess] /usr/local/lib/python3.7/site-packages/celery/fixups/django.py:202: UserWarning: Using settings.DEBUG leads to a memory leak, never use this setting in production environments!
  warnings.warn('Using settings.DEBUG leads to a memory leak, never '
[2023-02-15 19:12:38,470: INFO/MainProcess] celery@0d3e693c3dc8 ready.
```

And this is what happens when you make a change:

```
[2023-02-15 19:13:00,030: INFO/Beat] Task capdb.tasks.update_elasticsearch_from_queue[9a1ab4a9-44a2-49e5-b653-f142443b0d8d] succeeded in 0.025444916000196827s: None
Performing system checks...

Kill lingering celery worker...

worker: Warm shutdown (MainProcess)
Start celery worker...
System check identified no issues (0 silenced).
[2023-02-15 19:13:13,450: INFO/MainProcess] beat: Shutting down...
February 15, 2023 - 19:13:15
Django version 3.2.17, using settings 'config.settings'
Starting development server at http://0.0.0.0:8000/
Quit the server with CONTROL-C.

 -------------- celery@0d3e693c3dc8 v4.3.0 (rhubarb)
---- **** -----
--- * ***  * -- Linux-5.15.49-linuxkit-x86_64-with-debian-10.13 2023-02-15 19:13:23
-- * - **** ---
- ** ---------- [config]
- ** ---------- .> app:         config:0x4007f7b550
- ** ---------- .> transport:   redis://redis:6379//
- ** ---------- .> results:     redis://redis/
- *** --- * --- .> concurrency: 1 (prefork)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** -----
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery


[2023-02-15 19:13:23,783: INFO/MainProcess] Connected to redis://redis:6379//
[2023-02-15 19:13:23,850: INFO/MainProcess] mingle: searching for neighbors
[2023-02-15 19:13:24,967: INFO/MainProcess] mingle: all alone
[2023-02-15 19:13:25,012: WARNING/MainProcess] /usr/local/lib/python3.7/site-packages/celery/fixups/django.py:202: UserWarning: Using settings.DEBUG leads to a memory leak, never use this setting in production environments!
  warnings.warn('Using settings.DEBUG leads to a memory leak, never '
[2023-02-15 19:13:25,012: INFO/MainProcess] celery@0d3e693c3dc8 ready.
```

**Comment by rebeccacremona** [comment]
Looks good! Sorry for the slow followup review! I was surprised to see so many python processes, and spent longer than necessary trying to figure out what they all were / verifying that they all spin up and down as expected. (They do)

```
root@01d5910667c0:/app# ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.0 150552 12828 pts/0    Ssl+ 14:55   0:00 /usr/bin/qemu-x86_64 /bin/bash /bin/bash
root       105  0.0  0.0 150812 13276 pts/2    Ssl  15:18   0:00 /usr/bin/qemu-x86_64 /bin/bash bash
root       639  1.3  1.7 730052 285152 pts/1   Ssl+ 15:36   0:10 /usr/bin/qemu-x86_64 /usr/local/bin/python /usr/local/bin/python 
root       657  0.0  0.0 146540  8000 pts/1    Sl+  15:36   0:00 /usr/bin/qemu-x86_64 /bin/sh /bin/sh -c python manage.py run_cele
root       660  0.0  0.0 146540  6552 pts/1    Sl+  15:36   0:00 /usr/bin/qemu-x86_64 /bin/sh /bin/sh -c python manage.py runserve
root       663  1.5  1.7 732216 282168 pts/1   Sl+  15:36   0:11 /usr/bin/qemu-x86_64 /usr/local/bin/python python manage.py run_c
root       666  1.3  1.6 718512 270456 pts/1   Sl+  15:36   0:10 /usr/bin/qemu-x86_64 /usr/local/bin/python python manage.py runse
root       788 61.7  1.7 816336 294000 pts/1   Sl+  15:48   0:19 /usr/bin/qemu-x86_64 /usr/local/bin/python /usr/local/bin/python 
root       791 48.4  1.7 810940 286444 pts/1   Sl+  15:48   0:15 /usr/bin/qemu-x86_64 /usr/local/bin/python /usr/local/bin/python 
nobody     822 65.0  1.8 752652 302036 pts/1   Sl+  15:48   0:13 /usr/bin/qemu-x86_64 /usr/local/bin/python /usr/local/bin/python 
nobody     838  102  1.5 750596 258916 pts/1   Rl+  15:49   0:09 /usr/bin/qemu-x86_64 /usr/local/bin/python /usr/local/bin/python 
nobody     840  0.2  1.5 750916 259392 pts/1   Sl+  15:49   0:00 /usr/bin/qemu-x86_64 /usr/local/bin/python /usr/local/bin/python 
root       848  0.0  0.0 155904 10360 ?        Rl+  Feb18   0:00 ps aux
```

**Review by rebeccacremona — APPROVED** [review]

**Pull request merged**
